sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -985,6 +985,14 @@ object SparkSession extends Logging {
    * @since 2.0.0
    */
   def setActiveSession(session: SparkSession): Unit = {
+    if (SQLConf.get.legacyAllowModifyActiveSession) {
+      setActiveSessionInternal(session)
+    } else {
+      throw new IllegalStateException("Not allowed to modify active Spark session.")
```

Comment:
`UnsupportedOperationException` is better IMO

Reply:
done